elasticsearch/src/main/java/io/micronaut/elasticsearch/DefaultElasticsearchClientFactory.java

```diff
@@ -54,6 +60,18 @@ RestClient restClient(DefaultElasticsearchConfigurationProperties elasticsearchC
         return restClientBuilder(elasticsearchConfiguration).build();
     }
 
+    /**
+     * @param elasticsearchConfiguration The {@link DefaultElasticsearchConfigurationProperties} object
+     * @return The Elasticsearch Client
+     */
+    @Bean
+    ElasticsearchClient elasticsearchClient(DefaultElasticsearchConfigurationProperties elasticsearchConfiguration) {
+      RestClient restClient = restClientBuilder(elasticsearchConfiguration).build();
+
+      ElasticsearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
```

Comment:
The Elastic docs recommend using the same underlying low-level client during the transition to the new Java Client - 
https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/migrate-hlrc.html

Initially I was going to suggest only generating and sharing one `RestClient` object, but that might be a bigger refactor since `RestHighLevelClient` is _also_ generating its own unique `RestClient`.

Maybe the Micronaut devs (or one of the prior contributors) did it this way for a reason? Honestly not sure - I was wondering if it had something to do with supporting multiple Elasticsearch clusters, but Micronaut's `DefaultElasticsearchConfigurationProperties` doesn't actually support multiple clusters.

I say all that to say - your implementation seems to hold water to me, we just need to make sure the owners accept the tests. I'm happy to take a look at the groovy tests if they don't pass (once an owner kicks them off).

Reply:
Oh also - Micronaut provides an `ObjectMapper` bean - it's created in https://github.com/micronaut-projects/micronaut-core/blob/3.2.x/jackson-databind/src/main/java/io/micronaut/jackson/ObjectMapperFactory.java as described in https://docs.micronaut.io/3.2.4/guide/#jsonBinding .

We could theoretically inject that and use it here to construct the `JacksonJsonpMapper` or even take it a step further and make `JacksonJsonpMapper` its own bean. Really depends on if we want to use a new object mapper or use one provided by Micronaut

Reply:
This is / was a community contributed module, and is still maintained as such so I don't think the Micronaut devs have any particular opinion on what is best practise since we don't use Elastic search. 

What we do have to ensure is that if this module is changed it doesn't break existing apps and use cases. 

With regards to `ObjectMapper` we are trying to decouple of Jackson and have a new `JsonMapper` interface that is better to use.

Reply:
@MT-Jacobs I didn't want to mess around too much with the original implementation, that's why I reused the code. But I guess it's not a good design to do it this way.

@graemerocher Most libraries require an `ObjectMapper` instance these days, how do you plan to avoid this? Just have the modules add it themselves?

Reply:
See https://github.com/micronaut-projects/micronaut-rabbitmq/pull/293 for an example of how this is possible